src/functions/utils.php

```diff
@@ -595,138 +602,343 @@ function has_blocks( $post = null ) {
+
+				// '0' may be logged as a version number, which isn't useful, so we remove it
+				$history = array_filter( $history );
+				$history = array_unique( $history );
+
+				if ( ! empty( $history ) ) {
+					// Sort the array so smallest version number is first (likely how the array is stored anyway)
+					usort( $history, 'version_compare' );
+
+					return array_pop( $history );
+				}
+			}
+
+			// Fall back to the current plugin version.
+			if ( defined( get_class( $instance ) . '::VERSION' ) ) {
+				return $instance::VERSION;
+			}
+		}
 
-	// If no install version, let's assume it's been here a while.
-	if ( empty( $install_version ) ) {
+		// No version set.
 		return false;
 	}
+}
+
+if ( ! function_exists( 'tribe_get_first_ever_installed_version' ) ) {
+	/**
+	 * Gets the initially-recorded version number installed for the specified class of a plugin having a
+	 * `version_history_slug` property or a `VERSION` constant (i.e. Main classes).
+	 *
+	 * If user initially installed v2, downgraded to v1, then updated to v3, this will return v2.
+	 * If no historical version records, fallback is the class' current version.
+	 * If no version info found, it will return false.
+	 * Zero may have been logged as a past version but gets ignored.
+	 *
+	 * @since TBD
+	 *
+	 * @param string|object $class The plugin class' singleton name, class name, or instance.
+	 *
+	 * @return string|boolean The SemVer version string or false if no info found.
+	 */
+	function tribe_get_first_ever_installed_version( $class ) {
+		$instance = tribe_get_class_instance( $class );
+
+		if ( $instance ) {
+			// Try for the version history first.
+			if ( ! empty( $instance->version_history_slug ) ) {
+				$history = (array) Tribe__Settings_Manager::get_option( $instance->version_history_slug );
+
+				// '0' may be logged as a version number, which isn't useful, so we remove it
+				while (
+					! empty( $history )
+					&& empty( $history[0] )
+				) {
+					array_shift( $history );
+				}
+
+				// Found it so return it
+				if ( ! empty( $history[0] ) ) {
+					return $history[0];
+				}
+			}
+
+			// Fall back to the current plugin version.
+			if ( defined( get_class( $instance ) . '::VERSION' ) ) {
+				return $instance::VERSION;
+			}
+		}
 
-	return 0 < version_compare( $install_version, $version );
+		// No version set.
+		return false;
+	}
 }
 
-/**
- * Checks if a plugin was installed at/on the passed version.
- * If no info found, it will assume the plugin is old and return false.
- *
- * @since TBD
- *
- * @param object $class The plugin object.
- * @param string $version The SemVer version string.
- *
- * @return boolean Whether the plugin was installed at/on the passed version.
- */
-function tribe_installed_on( $class, $version ) {
-	$install_version = tribe_get_install_version( $class );
+if ( ! function_exists( 'tribe_get_currently_installed_version' ) ) {
```

Comment:
@sc0ttkclark `tribe_get_currently_installed_version()` is a better name for what I understood `tribe_get_latest_installed_version()` should be (last item on your checklist)

However, I don't really like it. I think it is only really that useful if we were to add "current version greater/less than" functions. Do we want those? If not, should I remove this function?

Reply:
I think there's still value in this function, we can keep it.

Reply:
@sc0ttkclark OK. Should I add "if current version is greater/less than" functions?

Reply:
I’m not sure we need it right away, but if you’d like to add it you can

Reply:
They're not _really_ necessary, since they'll both essentially be 
`version_compare( tribe_get_currently_installed_version( $class ), $version )`